src/sentry/static/sentry/app/views/groupDetails/shared/groupUserFeedback.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Link} from 'react-router';
+import {isEqual} from 'lodash';
+
+import SentryTypes from 'app/sentryTypes';
+import EventUserFeedback from 'app/components/events/userFeedback';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {t} from 'app/locale';
+import EmptyStateWarning from 'app/components/emptyStateWarning';
+import {Panel} from 'app/components/panels';
+import withOrganization from 'app/utils/withOrganization';
+import withApi from 'app/utils/withApi';
+
+class GroupUserFeedback extends React.Component {
+  static propTypes = {
+    organization: SentryTypes.Organization.isRequired,
+    group: SentryTypes.Group.isRequired,
+    query: PropTypes.object.isRequired,
+    api: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      error: false,
+      reportList: [],
+      pageLinks: '',
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!isEqual(prevProps.query, this.props.query)) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    this.setState({
+      loading: true,
+      error: false,
+    });
+
+    this.props.api.request(`/issues/${this.props.group.id}/user-reports/`, {
```

Comment:
This could be moved to an actionCreator function ⛳️

Reply:
I would like to avoid actionCreators unless we're updating global stores or doing some global action. In this case, the data is only used by this component. It's something I've seen practiced (very unofficially and not always consistently) throughout our codebase, but I think it's not a bad convention.

Reply:
I have been moving API calls into actionCreator modules incrementally to help keep components focused on rendering logic and ui state management. It also let's us solidate duplicate code around url generation and other request boilerplate.

In the future we may way to update those functions to update reflux stores but in the short term they return promises.